src/main/java/com/minelittlepony/hdskins/client/SkinUploader.java

```diff
@@ -108,8 +111,8 @@ public SkinServer getGateway() {
         return gateway;
     }
 
-    public boolean supportsFeature(Feature feature) {
-        return gateway != null && gateway.supportsFeature(feature);
+    public Set<Feature> getFeatures() {
+        return gateway != null ? gateway.getFeatures() : Collections.emptySet();
```

Comment:
Maybe use Optional here?

`gatway.map(SkinServer::getFeatures).orElseGet(Collections::emptySet)`

Reply:
I really don't want gateway to be nullable. It complicates things.

Reply:
It was nullable here in case someone configured the game with 0 skin servers. So at least then it doesn't cause this to crash when they try to use it.

Reply:
This class shouldn't be initialized if it's null.

Reply:
The uploader is initialised as part of the gui. It retrieves a gateway server from the active list of skin servers. If the active list of skin servers is empty, it gets a null gateway.